scripts/dev/e2e.py

```diff
@@ -156,14 +156,28 @@ def create_test_runner_pod(
             dev_config.namespace, field_selector=f"metadata.name=={TEST_RUNNER_NAME}",
         ),
         lambda pod_list: len(pod_list.items) == 0,
-        timeout=10,
+        timeout=30,
         sleep_time=0.5,
     ):
         raise Exception(
             "Execution timed out while waiting for the existing pod to be deleted"
         )
 
-    return corev1.create_namespaced_pod(dev_config.namespace, body=pod_body)
+    try:
+        corev1.create_namespaced_pod(dev_config.namespace, body=pod_body)
+    except ApiException as e:
+        if "No API token found for service account" in e.body:
+            print(
+                "API token was not found for service account, retrying in 10 seconds..."
+            )
+            time.sleep(10)
+            try:
+                corev1.create_namespaced_pod(dev_config.namespace, body=pod_body)
+            except ApiException as e:
+                print(
+                    "The API token was not found for two consecutive tries, aborting the test"
+                )
+                raise
```

Comment:
do you think we need to special case this specific error? This is generally just a case of the cluster not being ready for requests, maybe it would be simpler to just retry any ApiException a few times with a pause in between. (It's also not that obvious why we want to retry on this specific error twice, and not others). What do you think?

Reply:
Yeah I guess it makes sense. I didn't want to retry on exceptions that are caused by different issues but it's true that at the worst case they will happen again on the next try. I'll change it